src/queries.rs

```diff
@@ -23,8 +34,28 @@ pub fn find_queries_to_cache(
             AND digest_text NOT LIKE '%?=?%'
             AND s.sum_rows_sent > 0
             AND q.rule_id IS NULL
-            ORDER BY s.sum_rows_sent DESC",
-            config.source_hostgroup, config.readyset_user
+            ORDER BY s.sum_rows_sent DESC
+            LIMIT {}",
+            config.source_hostgroup,
+            config.readyset_user,
+            (config.number_of_queries * 2)
+        ))
+        .expect("Failed to query proxysql_conn");
```

Comment:
This error is now ambiguous; could make it more descriptive and say which query is failing.

Reply:
Good point. Fixed.